core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -129,13 +131,33 @@ angular.module('oppia').component('explorationSaveAndPublishButtons', {
         $scope.saveIsInProcess = false;
         $scope.publishIsInProcess = false;
         $scope.loadingDotsAreShown = false;
+        ctrl.saveBeforeUnload();
+      };
+
+      $scope.canPublish = false;
 
+      $scope.showPublishButton = function() {
         UserExplorationPermissionsService.getPermissionsAsync()
           .then(function(permissions) {
-            $scope.showPublishButton = function() {
-              return permissions.canPublish && $scope.isPrivate();
-            };
+            $scope.canPublish = permissions.canPublish;
+            $rootScope.$applyAsync();
```

Comment:
Why this was not required earlier?

Reply:
Why was what not required where?

Reply:
Comments appearing below a line of change generally refer to that line. In this case it is L143 (`$rootScope.$applyAsync()`).

Reply:
Clarified in Offline discussion